From 58eadb043036f0b6a9869bd200977a1990c7bf81 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Thu, 14 Oct 2021 13:40:33 +0300
Subject: [PATCH] [SW-56241] boot_info: restore back to sha256

Recently sha384 was added for gaudi2. As part of the modifications
for sha384, boot_info record in header was increased in size from 32
to 48, as if to adjust to be gaudi2 and and gaudi1. Yet, this
introduced backward compatibility issue, because reading of older
flash layout will not work correctly. This patch restore sha256
with boot_info (for both persistant and non-persistant).

Change-Id: I7cc60587cd45036d8bd913a765a6d91d0716a3e1
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/lib/preboot/boot_info.c b/lib/preboot/boot_info.c
index 3ed5d1e..72bbbee 100644
--- a/lib/preboot/boot_info.c
+++ b/lib/preboot/boot_info.c
@@ -9,7 +9,7 @@
 #include <preboot.h>
 #include <hl_boot_desc.h>
 #include <common.h>
-#include <sha.h>
+#include <sha256.h>
 
 #include "boot_info.h"
 #include "uart.h"
@@ -79,7 +79,7 @@
 	for (i = 0; i < boot_info.ptrs.count; i++) {
 		desc = &descs[i];
 		src = offset2ptr(desc->offset, PREBOOT_TEXT_BASE);
-		sha_csum((u8 *)src, desc->size, desc->chksum);
+		sha256_csum((u8 *)src, desc->size, desc->chksum);
 		copy_sz = ROUND_UP4(desc->size);
 
 		for (j = 0; j < copy_sz / sizeof(u32); j++)
@@ -169,7 +169,7 @@
 	struct hl_boot_info_ptrs ptrs;
 	struct hl_boot_info_desc desc;
 	u32 *offsets, i, cpy_sz;
-	u8 chksum[SHA_SUM_LEN];
+	u8 chksum[SHA256_SUM_LEN];
 	void *src;
 
 	ptrs = *(struct hl_boot_info_ptrs *)offset2ptr(0, BOOT_INFO_BASE);
@@ -192,8 +192,8 @@
 		src = offset2ptr(desc.offset, BOOT_INFO_BASE);
 		boot_info_memcpy32(data, src, cpy_sz);
 
-		sha_csum((u8 *)data, desc.size, chksum);
-		if (memcmp(chksum, desc.chksum, SHA_SUM_LEN))
+		sha256_csum((u8 *)data, desc.size, chksum);
+		if (memcmp(chksum, desc.chksum, SHA256_SUM_LEN))
 			return 1;
 
 		*copy_size = cpy_sz;
